sunpy/io/_file_tools.py

```diff
@@ -86,6 +86,8 @@ def _read(filepath, function_name, filetype=None, **kwargs):
         readername = detect_filetype(filepath)
         if readername not in _READERS.keys():
             readername = None
+        else:
+            return getattr(_READERS[readername], function_name)(filepath, **kwargs)
```

Comment:
I guess, could flip this logic
```suggestion
        if readername in _READERS.keys():
            return getattr(_READERS[readername], function_name)(filepath, **kwargs)
        
        readername = None            
```